### PR TITLE
Adds edge cases test code

### PR DESCRIPTION
**Pull request by jmjkim**
- Added testing codes for edge cases.

거침없는 feedback들 부탁할께 👀

### CODE REVIEW

File: src/list-pagination-context-provider.test.tsx [hunk 1]
```diff
@@ -90,26 +86,108 @@ describe('ListPaginationContextProvider', () => {
...
+          value={{
+            total: -4,
+            perPage: -2,
+          }}
+        >
+          <NaiveList />
+        </ListPaginationContextProvider>,
+      );
+
+      expect(getByText('currentPage: 1')).not.toBeNull();
```
Comment (morishjs) [inline src/list-pagination-context-provider.test.tsx]: 잘못된 input이 들어오면 잘못된 걸 고쳐서 보여주기 보다는 마찬가지로 No items to display가 보여지는게 좋을 것 같은데 어떻게 생각하니?

Comment (jmjkim) [reply src/list-pagination-context-provider.test.tsx]: 그렇네, 근본적으로 잘못된값을 받지않아야하는거니까, 수정할께!

**Review by morishjs — APPROVED** [review]
